dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MonitorMetricsHandler.java

```diff
@@ -0,0 +1,216 @@
+		        	Tr.debug(tc, "MBean Unregistered [" + objectName + "]");
+		            if (mappingTable.contains(objectName)) {
+		            	unregister(objectName);
+		            }
+		        }
+			}
+    	};
+    	
+		try {
+			MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+			mbs.addNotificationListener(MBeanServerDelegate.DELEGATE_NAME, listener, null, null);
+		} catch (InstanceNotFoundException e) {
+            if (tc.isDebugEnabled()) {
+                Tr.debug(tc, "getCount exception message: ", e.getMessage());
+                FFDCFilter.processException(e, getClass().getSimpleName(), "addMBeanListener:Exception");
+            }
+		}
+	}
+
+	protected void unregister(String objectName) {
+ 		Set<MonitorMetrics> removeSet = new HashSet<MonitorMetrics>();
+    	for (MonitorMetrics mm : metricsSet) {
+    		if (mm.getObjectName().equals(objectName)) {
+    			removeSet.add(mm);
+    			mm.unregisterMetrics(sharedMetricRegistry);
+    			Tr.debug(tc, "Monitoring MXBean " + objectName + " was unregistered from mpMetrics.");
+    		}
+    	}
+    	metricsSet.removeAll(removeSet);
+	}
+
+    protected void register() {
+        MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+        for (String sName : mappingTable.getKeys()) {
+            Set<ObjectInstance> mBeanObjectInstanceSet;
+            try {
+                mBeanObjectInstanceSet = mbs.queryMBeans(new ObjectName(sName), null);
+                if (sName.contains("ThreadPoolStats") && mBeanObjectInstanceSet.isEmpty() && execServ != null) {
+                    execServ.execute(() -> {
+                        final int MAX_TIME_OUT = 5000;
+                        int currentTimeOut = 0;
+                        Set<ObjectInstance> mBeanObjectInstanceSetTemp = mBeanObjectInstanceSet;
+                        while (mBeanObjectInstanceSetTemp.isEmpty() && currentTimeOut <= MAX_TIME_OUT) {
+                            try {
+                                Thread.sleep(50);
+
+                                mBeanObjectInstanceSetTemp = mbs.queryMBeans(new ObjectName(sName), null);
+                                currentTimeOut += 50;
+                            } catch (Exception e) {
+                                if (tc.isDebugEnabled()) {
+                                    Tr.debug(tc, "register exception message: ", e.getMessage());
+                                    FFDCFilter.processException(e, MonitorMetricsHandler.class.getSimpleName(),
+                                            "register:Exception");
+                                }
+                                /*
+                                 * Interruption Exception or RuntimeOperationException from malformed query exit
+                                 * thread.
+                                 */
+                                break;
+                            }
+                        }
+                        registerMbeanObjects(mBeanObjectInstanceSetTemp);
```

Comment:
When timed out, you will get an empty `mBeanObjectInstanceSetTemp`.  Do you need to handle that?